Update `minSdkVersion` (from `21` to `24`)

ParaskP7 opened the pull request:
This PR upgrades Utils to `minSdkVersion` to `24`. This has been unblocked because:

- `WPAndroid` was already on `minSdkVersion = 24`, for some time now (see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15135)).
- `WCAndroid` has been recently upgraded to `minSdkVersion = 24` as well (see [here](https://github.com/woocommerce/woocommerce-android/pull/7643)).

-----

PS: @JorgeMucientes @irfano I added you as the main reviewers, that is, in addition to the @wordpress-mobile/apps-infrastructure team itself, but randomly, since I just wanted someone from both, the `WordPress` and `Woo Commerce` teams, to sign-off on that change for WPAndroid and WCAndroid alike.

-----

### Testing instructions

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could:
    - Quickly smoke test, either the `WordPress` or `Woo Commerce` apps, with this version of `Utils`, and see if it works as expected.
    - Try and install, either the `WordPress` or `Woo Commerce` apps, with this version of `Utils`, on Android 5 (`API 21/22`) and Android 6 (`API 23`) devices and verify that you can't (see [API Levels](https://apilevels.com/)).

ParaskP7 committed:
Build: Extract common sdk versions to settings for all modules

This will now make sure that from now on all modules will be pointing to
the same 'min/compile/targetSdkVersion' versions, making such upgrades
much more straightforward as that will apply to all modules. This will
alleviate the risk of forgetting to update a specific module, or set of
modules due to an oversight.

FYI: Although this repo has only one module at the moment, this doesn't
mean that another module wouldn't be created in the future, for example
an 'example' app module. As such, this commit is firstly done as a best
practice, then so that all repos start becoming more consistent with
each other and finally as a preparation step for new module creation.

ParaskP7 committed:
Build: Update min sdk version to 24

A quick analysis showed that updating to 'minSdkVersion = 24' removed
just one 'OldTargetAPI' warning, without adding any new warnings.

irfano reviewed (approved):
LGTM! 👍🏻 
I smoke-tested the main features on WPAndroid and found no problem. minSdkVersion upgrade looks safe to me.

ParaskP7 commented:
Thank you so much for reviewing and testing this PR @irfano , you rock! 🙇 ❤️ 🚀

JorgeMucientes commented:
Thanks for handling this @ParaskP7 
Smoke-tested the WooCommerce app with this version of the library and looks good. Also tried installing it on an emulator with API 23 and couldn't. 

All good! :shipit:

ParaskP7 commented:
Thank you so much for reviewing and testing this PR @JorgeMucientes , you are awesome! 🙇 ❤️ 🚀

PS: And this is me 😊 , I should have said that one approval is enough, apologies for making you both do the same work, thus getting twice the approval. Then again, you can never be safer, right... 🤷 😝

ParaskP7 committed:
Build: Replace gradle ext with project ext for sdk versions declaration

For more info on that check PR comment and relevant discussion below:

https://github.com/wordpress-mobile/WordPress-FluxC-Android/
pull/2563#discussion_r1020079074
